backend/inspirehep/hal/tasks.py

```diff
@@ -72,13 +73,15 @@ def _hal_push(record):
         ids = record.get("external_system_identifiers", [])
         hal_value = get_values_for_schema(ids, "HAL")
         hal_id = hal_value[0] if hal_value else ""
-        if hal_id:
-            receipt = _hal_update(tei, hal_id, record)
-        else:
-            receipt = _hal_create(tei, record)
-        if receipt and receipt.id != hal_id:
-            _write_hal_id_to_record(record, receipt.id)
-        return receipt
+        lock_name = f"hal:{record['control_number']}"
+        with distributed_lock(lock_name, blocking=True):
```

Comment:
I don't understand why we need the lock here :/

Reply:
Because we want to prevent the race condition with disambiguation as @michamos wrote in the description in #2047